### ISSUE TITLE
where is model_test_one_image.py?

### ISSUE THREAD

**Issue by ma3252788** [pengyang1225/resnest-refinedet#2]
where is model_test_one_image.py?

**Comment by pengyang1225** [comment]
from __future__ import print_function
import torch

import torch.backends.cudnn as cudnn
from torch.autograd import Variable
from data import VOCAnnotationTransform, VOCDetection, BaseTransform
from models.refinedet import build_refinedet
# from models.refinedet_test import build_refinedet

import numpy as np
import cv2
import os
# import warnings
#
# warnings.filterwarnings("ignore")


def check_dir(dir):
    if not os.path.exists(dir):
        os.makedirs(dir)

if __name__ == '__main__':

    torch.set_default_tensor_type('torch.cuda.FloatTensor')

    num_classes = 2# len(labelmap) + 1                      # +1 for background
    net = build_refinedet(320, num_classes)            # initialize SSD
    net.load_state_dict(torch.load("/home/pengyang/Code/github/Refinedet_Pytorch-res86/test/RefineDet320_VOC_8000.pth"))
    net.eval()
    net = net.cuda()
    cudnn.benchmark = True

    print('Finished loading model!')


    img = cv2.imread("/home/pengyang/Code/hub_gitlab/YOLOv3-model-pruning/data/images/train/Buffy_3.jpg")
    h, w, channels = img.shape
    mean = np.array((104, 117, 123), dtype=np.float32)

    x = cv2.resize(img, (320, 320)).astype(np.float32)
    x -= mean
    x = x.astype(np.float32)
    x=torch.from_numpy(x).permute(2, 0, 1)
    x = Variable(x.unsqueeze(0))
    x = x.cuda()
    detections = net(x).data

    for j in range(1, detections.size(1)):
        dets = detections[0, j, :]
        mask = dets[:, 0].gt(0.).expand(5, dets.size(0)).t()
        dets = torch.masked_select(dets, mask).view(-1, 5)
        item = dets.cpu().numpy()
        if dets.size(0) == 0:
            continue
        boxes = dets[:, 1:]
        boxes[:, 0] *= w
        boxes[:, 2] *= w
        boxes[:, 1] *= h
        boxes[:, 3] *= h
        scores = dets[:, 0].cpu().numpy()
        cls_dets = np.hstack((boxes.cpu().numpy(), scores[:, np.newaxis])).astype(np.float32, copy=False)
        count = 0
        for k in item:
            if float(k[0]) >= 0.4:
                count += 1
                xmin = int(k[1] * w)
                ymin = int(k[2] * h)
                xmax = int(k[3] * w)
                ymax = int(k[4] * h)

                cv2.rectangle(img, (xmin, ymin), (xmax, ymax), (0, 0, 255), 2)
                cv2.rectangle(img, (xmin, ymin + 30), (xmin + 120, ymin + 2), (255, 128, 128), -1)
                cv2.putText(img, '1 : ' + '%.2f' % k[0], (xmin, ymin + 20), cv2.FONT_HERSHEY_SIMPLEX, 0.8, (0, 0, 255), 2, False)
    cv2.imwrite("result.jpg",img)

**Comment by pengyang1225** [comment]
from __future__ import print_function
import torch

import torch.backends.cudnn as cudnn
from torch.autograd import Variable
from data import VOCAnnotationTransform, VOCDetection, BaseTransform
from models.refinedet import build_refinedet
# from models.refinedet_test import build_refinedet

import numpy as np
import cv2
import os
# import warnings
#
# warnings.filterwarnings("ignore")


def check_dir(dir):
    if not os.path.exists(dir):
        os.makedirs(dir)

if __name__ == '__main__':

    torch.set_default_tensor_type('torch.cuda.FloatTensor')

    num_classes = 2# len(labelmap) + 1                      # +1 for background
    net = build_refinedet(320, num_classes)            # initialize SSD
    net.load_state_dict(torch.load("/home/pengyang/Code/github/Refinedet_Pytorch-res86/test/RefineDet320_VOC_8000.pth"))
    net.eval()
    net = net.cuda()
    cudnn.benchmark = True

    print('Finished loading model!')


    img = cv2.imread("/home/pengyang/Code/hub_gitlab/YOLOv3-model-pruning/data/images/train/Buffy_3.jpg")
    h, w, channels = img.shape
    mean = np.array((104, 117, 123), dtype=np.float32)

**Comment by pengyang1225** [comment]
`from __future__ import print_function
import torch

import torch.backends.cudnn as cudnn
from torch.autograd import Variable
from data import VOCAnnotationTransform, VOCDetection, BaseTransform
from models.refinedet import build_refinedet
# from models.refinedet_test import build_refinedet

import numpy as np
import cv2
import os
# import warnings
#
# warnings.filterwarnings("ignore")


def check_dir(dir):
    if not os.path.exists(dir):
        os.makedirs(dir)

if __name__ == '__main__':

    torch.set_default_tensor_type('torch.cuda.FloatTensor')

    num_classes = 2# len(labelmap) + 1                      # +1 for background
    net = build_refinedet(320, num_classes)            # initialize SSD
    net.load_state_dict(torch.load("/home/pengyang/Code/github/Refinedet_Pytorch-res86/test/RefineDet320_VOC_8000.pth"))
    net.eval()
    net = net.cuda()
    cudnn.benchmark = True

    print('Finished loading model!')


    img = cv2.imread("/home/pengyang/Code/hub_gitlab/YOLOv3-model-pruning/data/images/train/Buffy_3.jpg")
    h, w, channels = img.shape
    mean = np.array((104, 117, 123), dtype=np.float32)
`

**Comment by ma3252788** [comment]
Thank you very much!

When I was training, several losses were not displayed. What could be the reason for this? Can't I know mAP or accuracy during training? How to judge when to stop?

**Comment by ma3252788** [comment]
like this:
```
iter 83300 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5492 sec.
iter 83310 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5551 sec.
iter 83320 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5492 sec.
iter 83330 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5465 sec.
iter 83340 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5495 sec.
iter 83350 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5492 sec.
iter 83360 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5461 sec.
iter 83370 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5492 sec.
iter 83380 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5485 sec.
iter 83390 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5460 sec.
iter 83400 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5490 sec.
iter 83410 || ARM_L Loss: inf ARM_C Loss: 0.0000 ODM_L Loss: nan ODM_C Loss: 0.0122 || timer: 0.5496 sec.
```
